src/pages/Ticket/Component/Theater/Theater.js

```diff
@@ -1,49 +1,132 @@
   const [dataList, setDataList] = useState([]);
+  const [filterList, setFilterList] = useState([]);
+  const [timeList, setTimeList] = useState([]);
   const [loading, setLoading] = useState(true);
-  const [date, setDate] = useState(new Date());
+  const [newdate, setNewDate] = useState(new Date(Date.parse('2023-04-07')));
+  const { Thead } = Common;
+  const { movie, cinemaName, hallType, date, time, seat, totalPrice } =
+    selectInfo;
+
+  const ischecked =
+    selectInfo.movie === '' &&
+    selectInfo.region === '' &&
+    selectInfo.hallType === '' &&
+    selectInfo.cinemaName === '';
 
-  const { name, region, theater, cinema, time } = selectInfo;
+  const isVisible =
+    selectInfo.movie &&
+    selectInfo.region &&
+    selectInfo.cinemaName &&
+    selectInfo.hallType &&
+    selectInfo.date;
+
+  const options = {
+    year: 'numeric',
+    month: 'numeric',
+    day: 'numeric',
+    minimumIntegerDigits: 1,
+    useGrouping: false,
+  };
+
+  const formattedDate = newdate.toLocaleDateString('ko-KR', options);
+  const saveDate = formattedDate.replace(/\s|\.$/g, '');
 
   const handleChange = date => {
-    const formattedDate = date.toLocaleDateString('ko-KR');
-    setDate(date);
-    handleValue('date', formattedDate);
+    setNewDate(date);
+    handleValue('date', saveDate);
   };
 
   useEffect(() => {
-    fetch('/data/Theaterfilter.json')
-      .then(res => res.json())
-      .then(datas => {
-        setDataList(datas.data);
-        setLoading(false);
-      });
-  }, []);
+    const values = {
+      movie,
+      cinemaName,
+      hallType,
+      date,
+      time,
+      seat,
+    };
+
+    const removeSpaces = value => {
+      if (typeof value === 'string') {
+        return value.replace(/\s/g, '');
+      }
+      return value;
+    };
+
+    const sanitizedValues = Object.fromEntries(
+      Object.entries(values).map(([key, value]) => [key, removeSpaces(value)])
+    );
+
+    const queryParams = queryString.stringify(sanitizedValues, {
+      skipEmptyString: true,
+    });
+
+    ischecked
+      ? fetch(APIS.reservationAll)
+          .then(res => res.json())
+          .then(datas => {
+            setDataList(datas.getOptions);
+            setFilterList(datas.getOptions);
+            handleValue('date', saveDate.toString());
+            setLoading(false);
+          })
+      : fetch(`${APIS.reservationAll}?${queryParams}`)
+          .then(res => res.json())
+          .then(datas => {
+            setFilterList(datas.getOptions);
+            handleValue('date', saveDate);
+            setLoading(false);
+          });
+
+    isVisible &&
+      fetch(`${APIS.time}?${queryParams}`)
+        .then(res => res.json())
+        .then(datas => {
+          setTimeList(datas.getTimes);
+          setLoading(false);
+        });
+  }, [
+    ischecked,
+    saveDate,
+    handleValue,
+    isVisible,
+    movie,
+    cinemaName,
+    hallType,
+    date,
+    time,
+    seat,
+    totalPrice,
```

Comment:
```suggestion
selectInfo,
```
7개의 데이터는 결국 selectInfo에서 구조분해할당 한 값이기 때문에
이렇게 작성해도 될 것 같네요!